docs/core/tools/custom-templates.md

```diff
@@ -0,0 +1,232 @@
+
+The template engine is open source, and the online code repository is at [dotnet/templating](https://github.com/dotnet/templating/) on GitHub. You can open an issue if you run into a problem or start a discussion for a new feature, possibly a feature that you contribute. Visit the [dotnet/dotnet-template-samples](https://github.com/dotnet/dotnet-template-samples) repo for samples of templates. More templates, including templates from third parties, are found at [Available templates for dotnet new](https://github.com/dotnet/templating/wiki/Available-templates-for-dotnet-new) at GitHub. A detailed pre-release blog post that describes how to create and install custom templates written by [Sayed Ibrahim Hashimi](https://github.com/sayedihashimi) is at [How to create your own templates for dotnet new](https://blogs.msdn.microsoft.com/dotnet/2017/04/02/how-to-create-your-own-templates-for-dotnet-new/).
+
+## Configuration
+
+A template is composed of the following components:
+
+- Source files and folders
+- Configuration file (*template.json*)
+
+### Source files and folders
+
+The source files and folders are composed of whatever files and folders you want the template engine to use when the `dotnet new <TEMPLATE>` command is executed. The template engine is designed to use *runnable projects* as source code to produce projects. This has several benefits:
+
+- The template engine doesn't require you to inject special tokens into your project's source code.
+- The code files aren't special files or modified in any way to work with the template engine, so the tools you normally use when working with projects also work with template content.
+- You build, run, and debug your template projects just like you do for any of your other projects.
+- You can create a template from an existing project quickly by merely adding a *template.json* configuration file to the project.
+
+Files and folders stored in the template aren't limited to formal .NET project types, such as a .NET Core or .NET Framework solutions. Source files and folders may consist of any content that you wish to create when the template is used, even if the template engine produces just one file for its output. For example, you can create a template that contains a single *web.config* source file that creates a modified *web.config* file. The modifications to source files are based on logic and settings you've provided in the *template.json* configuration file along with values provided by the user passed as options to the `dotnet new <TEMPLATE>` command.
+
+### template.json
+
+The *template.json* file is placed in a *.template.config* folder in the root directory of the template. The file provides configuration to the template engine when creating output from the source files and folders. The minimum configuration requires the members shown in the following table, which is sufficient to add to a .NET Core app to produce a functional template.
```

Comment:
provides configuration -> provides configuration information

when creating output from the source files and folders -> is this necessary?